packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -47,3 +48,23 @@ voteCallbacks.castVoteAsync.add(async function updateNeedsReview (document: Vote
     void Users.update({_id:voter._id}, {$set:{needsReview: true}})
   }
 });
+
+voteCallbacks.castVoteAsync.add(async function handleDisableCommenting (document: VoteDocTuple) {
+  const author = await Users.findOne(document.vote.authorId);
+  const power = await Users.findOne(document.vote.power)
+  console.log("author karma: ", author?.karma, " author commenting disabled: ", author?.commentingDisabled)
+
+  // TODO: make it robust to unfortunate first comment
+  // Prevent users from commenting once their karma goes below a threshold
+  const threshold = -10
+  if (author) {
+    // BUG: this can fail for async reasons, in the case where the karma is updated before the author object is queried, and hence power has already been included
+    if (author?.karma <= threshold ) {
+      if (power + (author?.karma) > threshold) {
+        void Users.update({_id:author?._id}, {$set:{commentingDisabled: false}})
```

Comment:
Avoid using `?.` here. It won't actually happen because there's a null-check upstream, but if the _id were actually missing for some reason, this would turn into an `update` on `_id:undefined`, which would edit *every* user (if `multi:true` were passed, which it isn't), or one arbitrary user selected kind of at random (since it's a query with no constraints).